source/matrix_free/task_info.cc

```diff
@@ -36,6 +36,25 @@
 #include <iostream>
 #include <set>
 
+//
+// TBB with oneAPI API has deprecated and removed the
+// <code>tbb::tasks</code> backend. With this it is no longer possible to
+// compile the following code that builds a directed acyclic graph (DAG) of
+// (thread parallel) tasks without a major porting effort. It turned out
+// that such a dynamic handling of dependencies and structures is not as
+// competitive as initially assumed. Consequently, this part of the matrix
+// free infrastructure has seen less attention than the rest over the last
+// years and is (presumably) not used that often.
+//
+// In case of detected oneAPI backend we simply disable threading in the
+// matrix free backend for now.
+//
+// Matthias Maier, Martin Kronbichler, 2021
+//
+#ifdef DEAL_II_TBB_WITH_ONEAPI
+#  undef DEAL_II_WITH_TBB
+#endif
```

Comment:
You will want to re-enable the flag at the end of the file or you get in trouble if you do unity builds.

Reply:
Can we do
```cpp
#if defined(DEAL_II_WITH_TBB) && !defined(DEAL_II_TBB_WITH_ONEAPI)
```
instead of undefining?

Reply:
We can. `task_info.cc` is its own compilation unit so just undefining is fine.

Whatever you prefer - I feel agnostic about this :slightly_smiling_face:

Reply:
I support @drwells suggestion. This makes it clear in the place where it happens whether or not the TBB is used.

Reply:
As for unity builds: We shouldn't rely on the cmake unity build system putting `task_info.cc` into a compilation unit of its own. This will lead to strange problems if we just `#undef` something at the top -- which is why I prefer @drwells 's suggestion.